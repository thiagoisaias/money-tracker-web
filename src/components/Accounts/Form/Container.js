import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import Form from "./Form";

import {
  createAccount,
  updateAccount,
  setAccountToEdit,
  clearAccountsError,
  clearAccountToEdit
} from "store/actions/accounts/accounts";

export class Container extends Component {
  onSubmitData = formData => {
    const {
      accountToEdit,
      history,
      isLoading,
      match,
      onCreateAccount,
      onUpdateAccount
    } = this.props;

    if (isLoading) {
      return;
    }

    if (match.path === "/accounts/new") {
      onCreateAccount(formData, history);
    } else if (match.path === "/accounts/:id/edit" && accountToEdit) {
      onUpdateAccount(formData, accountToEdit.id, history);
    } else {
      return;
    }
  };

  render() {
    const { accountToEdit, error, isLoading, match } = this.props;
    return (
      <Form
        {...accountToEdit}
        onSubmitData={this.onSubmitData}
        error={error}
        isLoading={isLoading}
        match={match}
      />
    );
  }

  /* When the page is reloaded and the path is /account/:id/edit, accountToEdit is null, so the
  app redirects the user back to /accounts
  */
  componentDidMount() {
    const { accountToEdit, history, match } = this.props;
    if (match.path === "/accounts/:id/edit" && accountToEdit === null) {
      history.push("/accounts");
    }
  }

  // The value of accountToEdit should be null when the user leave /accounts/:id/edit
  componentWillUnmount() {
    const {
      error,
      match,
      onClearAccountsError,
      onClearAccountToEdit
    } = this.props;

    if (match.path === "/accounts/:id/edit") {
      onClearAccountToEdit();
    }

    if (error) {
      onClearAccountsError();
    }
  }
}

Container.propTypes = {
  accountToEdit: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    initialBalance: PropTypes.string.isRequired
  }),
  error: PropTypes.string,
  history: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  match: PropTypes.object.isRequired,
  onClearAccountToEdit: PropTypes.func.isRequired,
  onCreateAccount: PropTypes.func.isRequired,
  onUpdateAccount: PropTypes.func.isRequired,
  onSetAccountToEdit: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  accountToEdit: state.accounts.accountToEdit,
  isLoading: state.accounts.isLoading,
  error: state.accounts.error
});

const mapDispatchToProps = {
  onCreateAccount: createAccount,
  onSetAccountToEdit: setAccountToEdit,
  onClearAccountsError: clearAccountsError,
  onClearAccountToEdit: clearAccountToEdit,
  onUpdateAccount: updateAccount
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
