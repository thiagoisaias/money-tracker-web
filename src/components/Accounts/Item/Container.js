import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { accountType } from "types";

import Item from "./Item";

import {
  deleteAccount,
  setAccountToEdit
} from "store/actions/accounts/accounts";

export class Container extends Component {
  static propTypes = {
    accountData: accountType.isRequired,
    handleActiveItem: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    isActive: PropTypes.bool.isRequired,
    isLoading: PropTypes.bool.isRequired,
    deleteAccount: PropTypes.func.isRequired,
    setAccountToEdit: PropTypes.func.isRequired
  };

  handleEdit = () => {
    const { accountData, history, setAccountToEdit } = this.props;
    setAccountToEdit(accountData);
    history.push(`/accounts/${accountData.id}/edit`);
  };

  handleDelete = () => {
    const { accountData, isLoading, deleteAccount } = this.props;
    if (isLoading) {
      return;
    }

    if (!window.confirm("Are you sure you want to delete this account?")) {
      return;
    }
    deleteAccount(accountData.id);
  };

  render() {
    const { accountData, handleActiveItem, isActive } = this.props;
    return (
      <Item
        accountData={accountData}
        handleActiveItem={handleActiveItem}
        handleDelete={this.handleDelete}
        handleEdit={this.handleEdit}
        isActive={isActive}
      />
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.accounts.isLoading
});

const mapDispatchToProps = {
  deleteAccount,
  setAccountToEdit
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
