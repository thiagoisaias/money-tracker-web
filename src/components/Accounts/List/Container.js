import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { accountType } from "types";

import List from "./List";

import {
  fetchAccounts,
  clearAccountsError
} from "store/actions/accounts/accounts";

export class Container extends Component {
  static propTypes = {
    accountList: PropTypes.arrayOf(accountType).isRequired,
    error: PropTypes.string,
    isLoading: PropTypes.bool.isRequired,
    fetchAccounts: PropTypes.func.isRequired
  };

  componentDidMount() {
    const { fetchAccounts } = this.props;

    fetchAccounts();
  }

  render() {
    const { accountList, error, isLoading } = this.props;
    const propsToPass = {
      accountList,
      error,
      isLoading
    };
    return <List {...propsToPass} />;
  }

  componentWillUnmount() {
    const { error, clearAccountsError } = this.props;

    if (error) {
      clearAccountsError();
    }
  }
}

const mapStateToProps = state => ({
  isLoading: state.accounts.isLoading,
  error: state.accounts.error,
  accountList: state.accounts.accountList
});

const mapDispatchToProps = {
  fetchAccounts,
  clearAccountsError
};

export default connect(mapStateToProps, mapDispatchToProps)(Container);
