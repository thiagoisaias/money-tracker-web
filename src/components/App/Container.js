import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import App from "./App";

export class Container extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired
  };

  render() {
    return <App isAuthenticated={this.props.isAuthenticated} />;
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps)(Container));
