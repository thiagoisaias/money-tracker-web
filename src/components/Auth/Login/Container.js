import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { clearAuthError, login } from "store/actions/auth/auth";
import Form from "./Form";

export class Container extends Component {
  onSubmitData = formData => {
    const { history, isLoading, onLogin } = this.props;

    if (isLoading) {
      return;
    }

    onLogin(formData, history);
  };

  render() {
    const { onLogin, isLoading, error } = this.props;
    return (
      <Form
        error={error}
        isLoading={isLoading}
        onLogin={onLogin}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  componentWillUnmount() {
    const { error, onClearAuthError } = this.props;
    if (error) {
      onClearAuthError();
    }
  }
}

Container.propTypes = {
  error: PropTypes.string,
  onClearAuthError: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  error: state.auth.error
});

const mapDispatchToProps = dispatch => ({
  onClearAuthError: () => {
    dispatch(clearAuthError());
  },

  onLogin: (loginData, history) => {
    dispatch(login(loginData, history));
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
