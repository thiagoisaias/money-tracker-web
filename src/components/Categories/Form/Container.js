import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import {
  clearCategoryToEdit,
  createCategory,
  updateCategory
} from "store/actions/categories/categories";

import Form from "./Form";

export class Container extends Component {
  onSubmitData = formData => {
    const {
      categoryToEdit,
      history,
      isLoading,
      match,
      onCreateCategory,
      onUpdateCategory
    } = this.props;

    if (isLoading) {
      return;
    }

    if (match.path === "/categories/new") {
      onCreateCategory(formData, history);
    } else if (match.path === "/categories/:id/edit" && categoryToEdit) {
      onUpdateCategory(formData, categoryToEdit.id, history);
    } else {
      return;
    }
  };

  render() {
    const { categoryToEdit, error, isLoading, match } = this.props;
    return (
      <Form
        {...categoryToEdit}
        error={error}
        isLoading={isLoading}
        match={match}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  /* When the page is reloaded and the path is /category/:id/edit, categoryToEdit is null, so the
  app redirects the user back to /categories
  */
  componentDidMount() {
    const { categoryToEdit, history, match } = this.props;
    if (match.path === "/categories/:id/edit" && categoryToEdit === null) {
      history.push("/categories");
    }
  }

  // The value of categoryToEdit should be null when the user leave /categories/:id/edit
  componentWillUnmount() {
    const { match, onClearCategoryToEdit } = this.props;

    if (match.path === "/categories/:id/edit") {
      onClearCategoryToEdit();
    }
  }
}

Container.propTypes = {
  categoryToEdit: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  }),
  error: PropTypes.string,
  history: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  match: PropTypes.object.isRequired,
  onClearCategoryToEdit: PropTypes.func.isRequired,
  onCreateCategory: PropTypes.func.isRequired,
  onUpdateCategory: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  categoryToEdit: state.categories.categoryToEdit,
  error: state.categories.error,
  isLoading: state.categories.isLoading
});

const mapDispatchToProps = dispatch => ({
  onClearCategoryToEdit: () => {
    dispatch(clearCategoryToEdit());
  },

  onCreateCategory: (categoryData, history) => {
    dispatch(createCategory(categoryData, history));
  },

  onUpdateCategory: (categoryData, categoryId, history) => {
    dispatch(updateCategory(categoryData, categoryId, history));
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
