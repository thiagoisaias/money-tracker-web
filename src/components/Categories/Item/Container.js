import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { categoryType } from "types";

import {
  deleteCategory,
  setCategoryToEdit
} from "store/actions/categories/categories";

import Item from "./Item";

export class Container extends Component {
  static propTypes = {
    categoryData: categoryType.isRequired,
    handleActiveItem: PropTypes.func.isRequired,
    isActive: PropTypes.bool.isRequired,
    isLoading: PropTypes.bool.isRequired,
    deleteCategory: PropTypes.func.isRequired,
    setCategoryToEdit: PropTypes.func.isRequired
  };

  handleDelete = () => {
    const { categoryData, isLoading, deleteCategory } = this.props;

    if (isLoading) {
      return;
    }

    if (!window.confirm("Are you sure you want to delete this category?")) {
      return;
    }

    deleteCategory(categoryData.id);
  };

  handleEdit = () => {
    const { categoryData, history, setCategoryToEdit } = this.props;

    setCategoryToEdit(categoryData);
    history.push(`/categories/${categoryData.id}/edit`);
  };

  render() {
    const { categoryData, handleActiveItem, isActive } = this.props;

    return (
      <Item
        categoryData={categoryData}
        handleDelete={this.handleDelete}
        handleEdit={this.handleEdit}
        handleActiveItem={handleActiveItem}
        isActive={isActive}
      />
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.categories.isLoading
});

const mapDispatchToProps = {
  deleteCategory,
  setCategoryToEdit
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
