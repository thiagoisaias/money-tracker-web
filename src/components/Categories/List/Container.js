import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { fetchCategories } from "store/actions/categories/categories";

import List from "./List";

export class Container extends Component {
  render() {
    const { categoryList, error, isLoading } = this.props;
    return (
      <List categoryList={categoryList} error={error} isLoading={isLoading} />
    );
  }

  componentDidMount() {
    const { onFetchCategoryList } = this.props;
    onFetchCategoryList();
  }
}

Container.propTypes = {
  categoryList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      color: PropTypes.string.isRequired
    })
  ).isRequired,
  error: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  onFetchCategoryList: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    categoryList: state.categories.categoryList,
    error: state.categories.error,
    isLoading: state.categories.isLoading
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onFetchCategoryList: () => {
      dispatch(fetchCategories());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Container);
