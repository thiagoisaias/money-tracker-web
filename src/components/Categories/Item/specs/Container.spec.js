import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Container } from "../Container";

describe("Category Item Container", () => {
  const mockProps = {
    categoryData: {
      id: 1,
      name: "Lorem",
      color: "#A3H7S3"
    },
    handleActiveItem: jest.fn(),
    history: {},
    isActive: false,
    isLoading: false,
    onDeleteCategory: jest.fn(),
    onSetCategoryToEdit: jest.fn()
  };
  const component = shallow(<Container {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
