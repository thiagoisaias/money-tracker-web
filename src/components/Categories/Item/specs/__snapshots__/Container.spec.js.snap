// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Item Container renders properly 1`] = `
<Item
  categoryData={
    Object {
      "color": "#A3H7S3",
      "id": 1,
      "name": "Lorem",
    }
  }
  handleActiveItem={[Function]}
  handleDelete={[Function]}
  handleEdit={[Function]}
  isActive={false}
/>
`;
